### PR TITLE
[GNA] Add negative CNN2D tests

### PR DESCRIPTION
**Pull request by kbruniec**
### Details:
 - Add negative CNN2D tests

### Tickets:
 - 56129

### CODE REVIEW

File: inference-engine/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/convolution_negative.cpp [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+};
+
+/* ============= 2D Convolution ============= */
+const std::vector<std::vector<size_t >> kernels2D = {
+    {1, 3},
+    {7, 1},
+    {3, 3},
+};
+
+const std::vector<std::vector<size_t >> InvalidKernels2D = {
```
Comment (sirzabek) [inline inference-engine/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/convolution_negative.cpp:27]: Could be changed to kernels2DInvalid to align with other names.

File: inference-engine/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/convolution_negative.cpp [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+};
+const std::vector<std::vector<ptrdiff_t>> padBegins2DInvalid = { {1, 0}, {1, 1}, {0, 1}
+};
+const std::vector<std::vector<ptrdiff_t>> padEnds2DInvalid = { {1, 0}, {1, 1}, {0, 1}
+};
+const std::vector<std::vector<size_t >> dilations2D = { {1, 1},
+};
+const std::vector<std::vector<size_t >> dilations2DInvalid = { {2, 2},
+};
+const std::vector<size_t> numOutCannels2D = { 32 };
```
Comment (sirzabek) [inline inference-engine/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/convolution_negative.cpp:53]: Typo: should be Channels. Check in other places.

File: inference-engine/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/convolution_negative.cpp [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+                    ", expected: " << expected;
+            }
+        }
+    }
+    void SetUp() override {
+        ConvolutionLayerTest::SetUp();
+    }
+};
+
+#define GNA_NEG_INSTANTIATE(whats_wrong, sufix_params, sufix_input, error_message)                              \
```
Comment (sirzabek) [inline inference-engine/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/convolution_negative.cpp:145]: Typo: sufix->suffix.

**Review by sirzabek — APPROVED** [review]

File: inference-engine/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/convolution_negative.cpp (hunk 2 above)
Comment (kbruniec) [reply inference-engine/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/convolution_negative.cpp:53]: I will improve with the next PR

File: inference-engine/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/convolution_negative.cpp (hunk 1 above)
Comment (kbruniec) [reply inference-engine/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/convolution_negative.cpp:27]: I will improve with the next PR

File: inference-engine/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/convolution_negative.cpp (hunk 3 above)
Comment (kbruniec) [reply inference-engine/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/convolution_negative.cpp:145]: I will improve with the next PR

**Pull request merged**